azurerm/resource_arm_api_management_logger_test.go

```diff
@@ -0,0 +1,431 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMApiManagementLogger_basicEventHub(t *testing.T) {
+	resourceName := "azurerm_api_management_logger.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApiManagementLoggerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMApiManagementLogger_basicEventHub(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApiManagementLoggerExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "buffered", "true"),
+					resource.TestCheckResourceAttr(resourceName, "eventhub.#", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "eventhub.0.name"),
+					resource.TestCheckResourceAttrSet(resourceName, "eventhub.0.connection_string"),
+					resource.TestCheckNoResourceAttr(resourceName, "application_insights.#"),
```

Comment:
as this should be set to an empty array (above) we should be able to check there's 0 elements here:

```suggestion
					resource.TestCheckResourceAttr(resourceName, "application_insights.#", "0"),
```